src/components/PeopleFilters.tsx

```diff
@@ -1,12 +1,133 @@
+import { useSearchParams, Link, NavLink } from 'react-router-dom';
+
+enum Sex {
+  all = 'all',
+  mail = 'm',
+  female = 'f',
+}
+
+enum Params {
+  sex = 'sex',
+  query = 'query',
+  centuries = 'centuries',
+}
+
 export const PeopleFilters = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get(Params.sex) || '';
+  const query = searchParams.get(Params.query) || '';
+  const centuries = searchParams.getAll(Params.centuries) || '';
+
+  const handleChangeQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const params = new URLSearchParams(searchParams);
+
+    if (!event.target.value) {
+      params.delete(Params.query);
+      setSearchParams(params);
+
+      return;
+    }
+
+    params.set(Params.query, event.target.value);
+    setSearchParams(params);
+  };
+
+  const handleClickSex = (
+    event: React.MouseEvent<HTMLAnchorElement>,
+    currentSex: string,
+  ) => {
```

Comment:
```suggestion
    currentSex: Sex,
  ) => (event: React.MouseEvent<HTMLAnchorElement>) => {
```
after that you just can call this function in jsx.